### ISSUE TITLE
An AudioWaveGenerator Node to produce frequency sounds

### ISSUE THREAD

**Issue by KoreTeknology** [godotengine/godot-proposals#33]
### Describe the project you are working on:
The aim of this project is to add to the audio features a new type of node: Audio Wave generator, this node will produce a sound based on properties, related to audio synthesis.

### Describe how this feature / enhancement will help your project:
At the minimal level, this node will get some basic settings: Waveforms, ADSR, rate, pitch
on a secundary level, the enveloppe feature can be extended

### Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:
here is a preview:
![newnode](https://user-images.githubusercontent.com/14902346/64358899-e8f29200-cfcc-11e9-8588-ba80cdeba08b.jpg)

### If this enhancement will not be used often, can it be worked around with a few lines of script?:
Is there a reason why this should be core and not an add-on in the asset library?:

About the version of this, i would ask you to get some feedback about the choice to make an addon or a pure implementation in the audio core...please, let me know what you think...
i didnt started yet to implement this, but i am used to do this kind of work in C++, maybe i should start... anyway, this feature can be done an other way, apart from loading a wavfile, but still, without the parameters
Have a great day

**Comment by groud** [comment]
Sorry, there is an issue template, please it fill in.

**Comment by KoreTeknology** [comment]
hi groud... i just followed the guide... so where goes this kind of proposals ? please

**Comment by groud** [comment]
This kind of proposal can stay here, but you need to fill in the issue template, as it is asked on the top when you open an issue :
**Describe the project you are working on:**
**Describe how this feature / enhancement will help your project:**
**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
**Describe implementation detail for your proposal (in code), if possible:**
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
**Is there a reason why this should be core and not an add-on in the asset library?:**

**Comment by KoreTeknology** [comment]
ah ok, i was deleting the template to answer it properly and describe this project, i wasnt sure, thx for your support, i am modifying the request to fit your template, thanks to accept it

**Comment by groud** [comment]
Thanks :)

**Comment by reduz** [comment]
**Is there a reason why this should be core and not an add-on in the asset library?:**

**Comment by girng** [comment]
very cool, but might be better suited as an addon? not sure

**Comment by KoreTeknology** [comment]
You are right girng, now i look at it and for sure, as this must be done within much more features, and i (as many) want to have a clean code of godot core, i see here the oportunity to publish as an addon... let see the 3.2 audio updates and check for the next release what people think about audio "devices" and tools included, thx to all

**Comment by NoodleSushi** [comment]
I could try developing an add-on that does this; only downside is that playing on a very specific moment in time (syncing e.g. rhythm games) wouldn't be possible. It's gonna have a very short delay like any other `AudioStreamPlayer`s.

**Comment by RoseAndres** [comment]
I found this discussion looking to see of Godot already had something like this in the engine or something in the gdscript api to make the non-trivial task of coding waveforms easier.

I have a game I've been working on that has zones that affect the player character from some kind of waveform node/api would be the perfect fit for this.
